cms/djangoapps/maintenance/views.py

```diff
@@ -0,0 +1,231 @@
+            raise ItemNotFoundError(COURSE_KEY_ERROR_MESSAGES['course_key_not_found'])
+
+        # get branch specific locator
+        course_usage_key = course_usage_key.for_branch(branch)
+
+        return course_usage_key
+
+
+class ForcePublishCourseView(MaintenanceBaseView):
+    """
+    View for force publishing state of the course, used by the escalation team.
+    """
+
+    def __init__(self):
+        super(ForcePublishCourseView, self).__init__(MAINTENANCE_VIEWS['force_publish_course'])
+        self.context.update({
+            'current_versions': [],
+            'updated_versions': [],
+            'form_data': {
+                'course_id': '',
+                'is_dry_run': True
+            }
+        })
+
+    @transaction.atomic
+    @method_decorator(require_global_staff)
+    def post(self, request):
+        """
+        Force publishes a course.
+
+        Arguments:
+            course_id (string): a request parameter containing course id
+            is_dry_run (string): a request parameter containing dry run value.
+                                 It is obtained from checkbox so it has either values 'on' or ''.
+        """
+
+        course_id = request.POST.get('course-id')
+        is_dry_run = bool(request.POST.get('dry-run'))
+
+        self.context.update({
+            'form_data': {
+                'course_id': course_id,
+                'is_dry_run': is_dry_run
+            }
+        })
+
+        try:
+            course_usage_key = self.validate_course_key(course_id)
+        except InvalidKeyError:
+            self.context['error'] = True
+            self.context['msg'] = COURSE_KEY_ERROR_MESSAGES['invalid_course_key']
+        except ItemNotFoundError as exc:
+            self.context['error'] = True
+            self.context['msg'] = exc.message
+        except ValidationError as exc:
+            self.context['error'] = True
+            self.context['msg'] = exc.message
+
+        if self.context['error']:
+            return self.render_response()
+
+        source_store = modulestore()._get_modulestore_for_courselike(course_usage_key)  # pylint: disable=protected-access
+        if not hasattr(source_store, 'force_publish_course'):
+            self.context['msg'] = _("Force publish course does not support old mongo style courses.")
+            logging.info(
+                "Force publish course does not support old mongo style courses. \
+                %s attempted to force publish the course %s.",
+                request.user,
+                course_id,
+                exc_info=True
+            )
+            return self.render_response()
+
+        current_versions = get_course_versions(course_id)
+
+        # if publish and draft are NOT different
+        if current_versions['published-branch'] == current_versions['draft-branch']:
+            self.context['msg'] = _("Course is already in published state.")
+            logging.info(
```

Comment:
I am not sure `info` is the right thing here, `error` or `warning` should be considered.